### PR TITLE
web: Use SyntheticMouseEvent as a type for a value in mouse events listeners

### PR DESCRIPTION
**Pull request by eymar**

### CODE REVIEW

File: web/core/src/jsMain/kotlin/org/jetbrains/compose/web/events/SyntheticEvent.kt [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    val target: Element,
+    val nativeEvent: Event
+) {
+    val bubbles: Boolean = nativeEvent.bubbles
+    val cancelable: Boolean = nativeEvent.cancelable
+    val composed: Boolean = nativeEvent.composed
+    val currentTarget: HTMLElement? = nativeEvent.currentTarget.unsafeCast<HTMLInputElement?>()
+    val eventPhase: Short = nativeEvent.eventPhase
+    val defaultPrevented: Boolean = nativeEvent.defaultPrevented
+    val timestamp: Number = nativeEvent.timeStamp
```
Comment (Skolotsky) [inline web/core/src/jsMain/kotlin/org/jetbrains/compose/web/events/SyntheticEvent.kt:18]: I think it should be Int

File: web/core/src/jsMain/kotlin/org/jetbrains/compose/web/events/SyntheticMouseEvent.kt [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+    constructor(nativeEvent: WheelEvent): this(
+        target = nativeEvent.target as HTMLElement,
+        nativeEvent = nativeEvent
+    )
+
+    val deltaX: Double = nativeEvent.deltaX
+    val deltaY: Double = nativeEvent.deltaY
+    val deltaZ: Double = nativeEvent.deltaZ
+    val deltaMode: Int = nativeEvent.deltaMode
```
Comment (Skolotsky) [inline web/core/src/jsMain/kotlin/org/jetbrains/compose/web/events/SyntheticMouseEvent.kt:69]: Is it sort of enum?

**Review by Skolotsky — CHANGES_REQUESTED** [review]

**Review by Schahen — APPROVED** [review]

File: web/core/src/jsMain/kotlin/org/jetbrains/compose/web/events/SyntheticMouseEvent.kt (hunk 2 above)
Comment (eymar) [reply web/core/src/jsMain/kotlin/org/jetbrains/compose/web/events/SyntheticMouseEvent.kt:69]: yes, sort of:

here is its defenition in kotlin/js
```kotlin
 companion object {
        val DOM_DELTA_PIXEL: Int
        val DOM_DELTA_LINE: Int
        val DOM_DELTA_PAGE: Int
        val NONE: Short
        val CAPTURING_PHASE: Short
        val AT_TARGET: Short
        val BUBBLING_PHASE: Short
    }
```

File: web/core/src/jsMain/kotlin/org/jetbrains/compose/web/events/SyntheticEvent.kt (hunk 1 above)
Comment (eymar) [reply web/core/src/jsMain/kotlin/org/jetbrains/compose/web/events/SyntheticEvent.kt:18]: in kotlin/js nativeEvent the type is Number, so I followed its example.

Did you want to suggest that it's better to cast to Int?

File: web/core/src/jsMain/kotlin/org/jetbrains/compose/web/events/SyntheticMouseEvent.kt (hunk 2 above)
Comment (Skolotsky) [reply web/core/src/jsMain/kotlin/org/jetbrains/compose/web/events/SyntheticMouseEvent.kt:55]: it's better to use another type than `Int`, to prompt available values and use safe `when(deltaMode)`

File: web/core/src/jsMain/kotlin/org/jetbrains/compose/web/events/SyntheticEvent.kt (hunk 1 above)
Comment (Skolotsky) [reply web/core/src/jsMain/kotlin/org/jetbrains/compose/web/events/SyntheticEvent.kt:16]: It's bad typing, timestamp is never Double

**Pull request merged**